### ISSUE TITLE
Add automatic semver check to CI

### ISSUE THREAD

**Issue by haerdib** [scs/substrate-api-client#527]
According to the comment of @Niederb here: https://github.com/scs/substrate-api-client/pull/515#pullrequestreview-1372403763:
> I recently saw this [cargo-semver-checks](https://crates.io/crates/cargo-semver-checks) maybe this would make it easier to identify breaking changes. I haven't tried it though.

I quickly looked into it, but it seems cargo semver, while it does support semver check against github revisions with cli, it does not support this in Github Actions. It seems to only check against crates.io. 
To solve this issue we can either
- use another semver checker
- publish on crates.io

**Comment by Niederb** [comment]
I agree that publishing on crates.io is probably the easiest approach but did you see the section "Does the crate I'm checking have to be published on crates.io?" in the readme?

**Comment by haerdib** [comment]
Yes, I saw that. But as far as I understood, that's for the cli version. For the GitHub Action one, I didn't find that option. Ofc, one could build the cli version into GHA, but I'm not sure if that's worth it.

**Comment by Niederb** [comment]
> Yes, I saw that. But as far as I understood, that's for the cli version. For the GitHub Action one, I didn't find that option. Ofc, one could build the cli version into GHA, but I'm not sure if that's worth it.

Ok, I just wanted to mention it. You already gave it more thought than I did 👍
